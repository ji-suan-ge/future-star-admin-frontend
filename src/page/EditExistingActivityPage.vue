<template>
  <el-container>
    <el-main>
      <el-page-header @back="goBack()" content="修改活动详情"></el-page-header>
      <el-divider/>
      <el-card class="activity-card" shadow="always">
        <el-form class="activity-table" label-width="150px">
          <el-form-item class="activity-title" label="活动标题：">
            <el-input class="short-text"/>
          </el-form-item>
          <el-form-item label="活动项目：">
            <el-input class="short-text"/>
          </el-form-item>
          <el-row>
            <el-col :span="12">
              <el-form-item label="活动时间：">
                <el-date-picker
                  v-model="value1"
                  type="datetime"
                  placeholder="选择日期时间">
                </el-date-picker>
              </el-form-item>
            </el-col>
            <el-col :span="12">
              <el-form-item label="活动地点：">
                <el-input class="short-text"/>
              </el-form-item>
            </el-col>
          </el-row>
          <el-row>
            <el-col :span="8">
              <el-form-item label="面向人群：">
                <el-input class="short-text"/>
              </el-form-item>
            </el-col>
            <el-col :span="8">
              <el-form-item label="限制人数：">
                <el-input class="short-text"/>
              </el-form-item>
            </el-col>
            <el-col :span="8">
              <el-form-item label="收费情况：">
                <el-input class="short-text"/>
              </el-form-item>
            </el-col>
          </el-row>
          <el-form-item label="活动内容描述：">
            <el-input class="long-text" type="textarea"/>
          </el-form-item>
          <el-form-item label="组织单位信息：">
            <el-input class="long-text" type="textarea"/>
          </el-form-item>
          <el-form-item label="组织单位联系方式：">
            <el-input class="long-text" type="textarea"/>
          </el-form-item>
          <el-form-item label="报名截止时间：">
            <el-date-picker
              v-model="value1"
              type="datetimerange"
              start-placeholder="开始日期"
              end-placeholder="结束日期"
              :default-time="['12:00:00']">
            </el-date-picker>
          </el-form-item>
          <el-form-item>
            <el-button type="primary">确认修改</el-button>
            <el-button>取消</el-button>
          </el-form-item>
        </el-form>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'EditExistingActivityPage',
  methods: {
    goBack: function () {
      this.$router.go(-1)
    }
  }
}
</script>

<style scoped>
  .activity-card {
    width: 70%;
    margin: auto;
  }

  .activity-table {
    margin: 40px 20px;
  }
</style>
