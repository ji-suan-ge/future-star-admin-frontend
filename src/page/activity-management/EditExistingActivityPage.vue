<template>
  <el-container>
    <el-main>
      <el-page-header @back="goBack()" content="修改校友活动"></el-page-header>
      <el-divider/>
      <el-card class="activity-card" shadow="never">
        <el-row>
          <el-col
            :span="24"
            class="center-position">
            <div class="block">
              <el-upload
                class="avatar-uploader"
                action="https://jsonplaceholder.typicode.com/posts/"
                :show-file-list="false"
                :http-request="uploadImg"
                :on-success="uploadIconSuccess"
                :on-remove="handleRemove">
                <img width="140px" :src="currentActivity.icon" class="clazz-image" alt="avatar">
              </el-upload>
            </div>
          </el-col>
        </el-row>
        <div class="line">
          <el-divider></el-divider>
        </div>
        <div class="description">- 活动信息 -</div>
        <el-form class="activity-table" label-width="150px">
          <el-row class="top-distance">
            <el-col :span="8" :offset="offset">
              <i class="el-icon-tickets">&nbsp;活动名称：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.name"
                maxlength="20"
                show-word-limit>
                <template slot="prepend">名称:</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-watch">&nbsp;报名开始时间：</i>
              <el-date-picker
                v-model="currentActivity.enroll_start_time"
                type="date"
                placeholder="请选择开始时间">
                <template slot="prepend">开始时间：</template>
              </el-date-picker>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-watch">&nbsp;报名结束时间：</i>
              <el-date-picker
                v-model="currentActivity.enroll_end_time"
                type="date"
                placeholder="请选择结束时间">
                <template slot="prepend">结束时间：</template>
              </el-date-picker>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-watch">&nbsp;活动时间：</i>
              <el-date-picker
                v-model="currentActivity.start_time"
                type="date"
                placeholder="请选择日期时间">
                <template slot="prepend">活动时间：</template>
              </el-date-picker>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-user">&nbsp;面向人群：</i>
              <el-select
                v-model="clazz"
                collapse-tags
                :multiple="true"
                placeholder="请选择">
                <el-option v-for="item in clazzOptions"
                           :key="item.clazz"
                           :label="item.label"
                           :value="item.clazz">
                </el-option>
              </el-select>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-location-information">&nbsp;活动地点：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.address"
                maxlength="20"
                show-word-limit>
                <template slot="prepend">城市：</template>
              </el-input>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-crop">&nbsp;限制人数：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.people_number_limit"
                maxlength="3"
                show-word-limit>
                <template slot="prepend">人数：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="6" :offset="offset">
              <i class="el-icon-money">&nbsp;收费情况：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.price"
                maxlength="4"
                show-word-limit>
                <template slot="prepend">金额(¥)：</template>
              </el-input>
            </el-col>
            <el-col :span="6" :offset="2">
              <i class="el-icon-s-finance">&nbsp;主办方：</i>
              <el-input
                type="text"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.organizer"
                maxlength="50"
                show-word-limit>
                <template slot="prepend">名称：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <el-col :span="14" :offset="offset">
              <i class="el-icon-fork-spoon">&nbsp;活动日程：</i>
              <el-input
                type="textarea"
                size="medium"
                placeholder="请输入内容"
                v-model="currentActivity.arrangement"
                maxlength="100"
                show-word-limit
                :autosize="{ minRows: 4, maxRows: 8}">
                <template slot="prepend">活动日程：</template>
              </el-input>
            </el-col>
          </el-row>
          <el-row class="top-distance">
            <div class="block">
              <el-upload
                align="center"
                class="avatar-uploader"
                action="https://jsonplaceholder.typicode.com/posts/"
                :show-file-list="false"
                :http-request="uploadImg"
                :on-success="uploadImgSuccess"
                :on-remove="handleRemove">
                <img width="400px" :src="currentActivity.image" class="clazz-image" alt="avatar">
              </el-upload>
            </div>
          </el-row>
          <el-row>
            <el-col :span="24" class="button">
              <el-button type="primary" @click="editActivity()">确认修改</el-button>
              <el-button @click="goBack()">取消</el-button>
            </el-col>
          </el-row>
        </el-form>
      </el-card>
    </el-main>
  </el-container>
</template>

<script>
export default {
  name: 'EditExistingActivityPage',
  data () {
    return {
      page_size: 20,
      currentPage: 1,
      offset: 5,
      clazz: [],
      oldClazz: [],
      currentActivity: [],
      clazzOptions: []
    }
  },
  created () {
    this.currentActivity = this.$store.getters.getCurrentActivity
    if (this.axios) {
      this.findAllClasses()
      this.findExistClasses()
    }
  },
  methods: {
    goBack () {
      this.$router.go(-1)
    },
    findAllClasses () {
      let that = this
      let params = {
        page: that.currentPage,
        page_size: that.page_size
      }
      that.axios.get('/clazz/clazz', { params }).then((response) => {
        if (response.data.code === '2000') {
          let results = response.data.data.results
          for (let i = 0; i < results.length; i++) {
            that.clazzOptions.push({
              clazz: results[i]['id'],
              label: results[i]['name']
            })
          }
        }
        that.$forceUpdate()
      })
    },
    findExistClasses () {
      let that = this
      let params = {
        page: that.currentPage,
        page_size: that.page_size,
        activity_id: that.currentActivity.id
      }
      that.axios.get('/activity/clazz', { params }).then((response) => {
        if (response.data.code === '2000') {
          let activityResults = response.data.data.results
          for (let i = 0; i < activityResults.length; i++) {
            that.clazz.push(activityResults[i].id)
            that.oldClazz.push(activityResults[i].id)
          }
        }
        that.$forceUpdate()
      })
    },
    uploadImg (f) {
      let param = new FormData()
      let url = '/file/upload'
      param.append('file', f.file)
      let config = {
        headers: { 'Content-Type': 'multipart/form-data' }
      }
      this.axios.post(url, param, config)
        .then(response => {
          f.onSuccess(response.data)
        })
        .catch(() => {
          f.onError()
        })
    },
    uploadImgSuccess (response, file, fileList) {
      this.currentActivity.image = response.data.url
      this.$message({
        showClose: true,
        message: '修改活动图片成功',
        type: 'success' })
    },
    uploadIconSuccess (response, file, fileList) {
      this.currentActivity.icon = response.data.url
      this.$message({
        showClose: true,
        message: '修改活动图标成功',
        type: 'success' })
    },
    handleRemove (file, fileList) {
      console.log('文件删除')
    },
    editActivity () {
      let params = {
        name: this.currentActivity.name,
        enroll_start_time: this.currentActivity.enroll_start_time.replace(/\//g, '-'),
        enroll_end_time: this.currentActivity.enroll_end_time.replace(/\//g, '-'),
        organizer: this.currentActivity.organizer,
        start_time: this.currentActivity.start_time.replace(/\//g, '-'),
        address: this.currentActivity.address,
        arrangement: this.currentActivity.arrangement,
        price: this.currentActivity.price,
        people_number_limit: this.currentActivity.people_number_limit
      }
      let url = '/activity/activity/' + this.currentActivity.id
      this.axios.put(url, params).then((response) => {
        if (response.data.code === '2000') {
          this.$message({
            type: 'success',
            message: '修改活动成功',
            duration: 2000
          })
          let url = '/activity/clazz/' + this.currentActivity.id + '/all'
          this.axios.delete(url).then(() => {
            for (let i = 0; i < this.clazz.length; i++) {
              this.axios.post('/activity/clazz', {
                activity_id: this.currentActivity.id,
                clazz_id: this.clazz[i]
              }).then((response) => {
                if (response.data.code === '2000') {
                  this.$message({
                    type: 'success',
                    message: '班级关联活动成功',
                    duration: 2000
                  })
                } else {
                  this.$message({
                    type: 'error',
                    message: '请求出错',
                    duration: 2000
                  })
                }
              }).catch((error) => {
                console.log(error)
                this.$message({
                  type: 'error',
                  message: '网络繁忙，请稍后重试',
                  duration: 2000
                })
              })
            }
          })
        } else {
          this.$message({
            type: 'error',
            message: '网络繁忙，请稍后重试',
            duration: 2000
          })
        }
      }).catch((error) => {
        console.log(error)
        this.$message({
          type: 'error',
          message: '网络繁忙',
          duration: 2000
        })
      })
    }
  }
}
</script>

<style scoped>
  .activity-table {
    margin: 40px 20px;
  }

  .description {
    font-size: 21px;
    text-align: center;
    color: black;
  }

  .center-position {
    text-align: center;
  }

  .el-col > i {
    font-size: 17px;
    color: #272324;
  }

  .el-col > input {
    font-size: 15px;
  }

  .el-col > label {
    margin-bottom: 100px;
    font-size: 14px;
  }

  .top-distance {
    margin-top: 40px;
  }

  .button {
    margin-top: 30px;
    text-align: center;
  }
</style>
