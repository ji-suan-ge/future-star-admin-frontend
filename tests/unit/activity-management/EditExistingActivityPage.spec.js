import { mount } from '@vue/test-utils'
import EditExistingActivityPage from '../../../src/page/activity-management/EditExistingActivityPage'

describe('EditExistingActivityPage', () => {
  const wrapper = mount(EditExistingActivityPage)
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
