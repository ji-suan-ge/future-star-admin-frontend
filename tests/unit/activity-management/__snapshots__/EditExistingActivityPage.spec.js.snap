// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExistingActivityPage 快照测试 1`] = `
<section class="el-container">
  <main class="el-main">
    <div class="el-page-header">
      <div class="el-page-header__left"><i class="el-icon-back"></i>
        <div class="el-page-header__title">返回</div>
      </div>
      <div class="el-page-header__content">修改校友活动</div>
    </div>
    <div class="el-divider el-divider--horizontal">
      <!---->
    </div>
    <div class="el-card activity-card is-never-shadow">
      <!---->
      <div class="el-card__body">
        <div class="el-row">
          <div class="center-position el-col el-col-24">
            <div class="block"><span class="el-avatar el-avatar--circle" style="height: 130px; width: 130px; line-height: 130px;"></span></div>
          </div>
        </div>
        <div class="line">
          <div class="el-divider el-divider--horizontal">
            <!---->
          </div>
        </div>
        <div class="description">- 活动信息 -</div>
        <form class="el-form activity-table">
          <div class="top-distance el-row">
            <div class="el-col el-col-8 el-col-offset-5"><i class="el-icon-tickets">&nbsp;活动名称：</i>
              <div class="el-input el-input--medium el-input-group el-input-group--prepend">
                <div class="el-input-group__prepend">活动名称:</div><input type="text" autocomplete="off" placeholder="请输入内容" maxlength="20" class="el-input__inner">
                <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><!----><!----><!----><span class="el-input__count"><span class="el-input__count-inner">
            0/20
          </span></span></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-6 el-col-offset-5"><i class="el-icon-watch">&nbsp;报名开始时间：</i>
              <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date">
                <!----><input type="text" autocomplete="off" name="" placeholder="请选择开始时间" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-date"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----><!----><!----><!----></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
            <div class="el-col el-col-6 el-col-offset-2"><i class="el-icon-watch">&nbsp;报名结束时间：</i>
              <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date">
                <!----><input type="text" autocomplete="off" name="" placeholder="请选择结束时间" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-date"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----><!----><!----><!----></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-6 el-col-offset-5"><i class="el-icon-watch">&nbsp;活动时间：</i>
              <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date">
                <!----><input type="text" autocomplete="off" name="" placeholder="请选择日期时间" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-date"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----><!----><!----><!----></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
            <div class="el-col el-col-6 el-col-offset-2"><i class="el-icon-user">&nbsp;面向人群：</i>
              <div class="el-select">
                <div class="el-select__tags" style="max-width: -32px; width: 100%;">
                  <!---->
                  <!---->
                  <!---->
                </div>
                <div class="el-input el-input--suffix">
                  <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择" class="el-input__inner">
                  <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
                  <!----></span>
                  <!---->
                  <!---->
                </div>
                <div class="el-select-dropdown el-popper is-multiple" style="display: none;">
                  <div class="el-scrollbar" style="display: none;">
                    <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                      <ul class="el-scrollbar__view el-select-dropdown__list"></ul>
                    </div>
                    <div class="el-scrollbar__bar is-horizontal">
                      <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                    </div>
                    <div class="el-scrollbar__bar is-vertical">
                      <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                    </div>
                  </div>
                  <p class="el-select-dropdown__empty">
                    无数据
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-6 el-col-offset-5"><i class="el-icon-location-information">&nbsp;活动地点：</i>
              <div class="el-input el-input--medium el-input-group el-input-group--prepend">
                <div class="el-input-group__prepend">活动地点：</div><input type="text" autocomplete="off" placeholder="请输入内容" maxlength="20" class="el-input__inner">
                <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><!----><!----><!----><span class="el-input__count"><span class="el-input__count-inner">
            0/20
          </span></span></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
            <div class="el-col el-col-6 el-col-offset-2"><i class="el-icon-crop">&nbsp;限制人数：</i>
              <div class="el-input el-input--medium el-input-group el-input-group--prepend">
                <div class="el-input-group__prepend">限制人数：</div><input type="text" autocomplete="off" placeholder="请输入内容" maxlength="3" class="el-input__inner">
                <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><!----><!----><!----><span class="el-input__count"><span class="el-input__count-inner">
            0/3
          </span></span></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-6 el-col-offset-5"><i class="el-icon-money">&nbsp;收费情况：</i>
              <div class="el-input el-input--medium el-input-group el-input-group--prepend">
                <div class="el-input-group__prepend">收费情况：</div><input type="text" autocomplete="off" placeholder="请输入内容" maxlength="4" class="el-input__inner">
                <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><!----><!----><!----><span class="el-input__count"><span class="el-input__count-inner">
            0/4
          </span></span></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
            <div class="el-col el-col-6 el-col-offset-2"><i class="el-icon-s-finance">&nbsp;主办方：</i>
              <div class="el-input el-input--medium el-input-group el-input-group--prepend">
                <div class="el-input-group__prepend">主办方：</div><input type="text" autocomplete="off" placeholder="请输入内容" maxlength="50" class="el-input__inner">
                <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><!----><!----><!----><span class="el-input__count"><span class="el-input__count-inner">
            0/50
          </span></span></span>
                <!----></span>
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <div class="top-distance el-row">
            <div class="el-col el-col-14 el-col-offset-5"><i class="el-icon-fork-spoon">&nbsp;活动日程：</i>
              <div class="el-textarea el-input--medium el-input-group el-input-group--prepend"><textarea autocomplete="off" placeholder="请输入内容" maxlength="100" class="el-textarea__inner" style="min-height: -8px; height: -16px;"></textarea><span class="el-input__count">0/100</span></div>
            </div>
          </div>
          <div class="el-row">
            <div class="button el-col el-col-24"><button type="button" class="el-button el-button--primary">
                <!---->
                <!----><span>确认修改</span></button> <button type="button" class="el-button el-button--default">
                <!---->
                <!----><span>取消</span></button></div>
          </div>
        </form>
      </div>
    </div>
  </main>
</section>
`;
